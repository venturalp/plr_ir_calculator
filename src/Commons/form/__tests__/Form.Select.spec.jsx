import {
  renderWithRouterDocument,
  ThemeWrapper,
} from 'Commons/tests/Tests.Helpers'
import React from 'react'
import { FormSelect } from '../Form.Select'

const options = [
  { id: 1, name: 'foo' },
  { id: 2, name: 'bar' },
]

describe('test form select', () => {
  it('test single select', () => {
    const { asFragment } = renderWithRouterDocument(
      <ThemeWrapper>
        <FormSelect options={options} value={1} />
      </ThemeWrapper>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
