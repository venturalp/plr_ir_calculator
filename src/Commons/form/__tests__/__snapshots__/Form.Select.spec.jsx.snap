// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form select test single select 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="formSelect"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="formSelect"
        role="button"
        tabindex="0"
      >
        foo
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        name="formSelect"
        tabindex="-1"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-2"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root select-error MuiFormHelperText-filled"
    />
  </div>
</DocumentFragment>
`;
